agent/internal/actor/http.go

```diff
@@ -37,8 +38,14 @@ func NewIPActionHTTPHandler(action Action, ip net.IP) (http.Handler, error) {
 
 // NewUserActionHTTPHandler returns a HTTP handler that should be applied at
 // the request handler level to perform the security response.
-func NewUserActionHTTPHandler(action Action, userID map[string]string) http.Handler {
-	return newBlockHTTPHandler(blockUserEventName, newBlockedUserEventProperties(action, userID))
+func NewUserActionHTTPHandler(action Action, userID map[string]string) (http.Handler, error) {
```

Comment:
Maybe add a dedicated named type for this `map[string]string`. Also `userID` arg feels wrong if there can be many. `userIDs`, or even just `users`.

Reply:
It really is just one user, uniquely identified by a set of key values.